src/parser/expression/constant_value_expression.cpp

```diff
@@ -0,0 +1,356 @@
+#include "parser/expression/constant_value_expression.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "common/hash_util.h"
+#include "execution/sql/value.h"
+#include "execution/sql/value_util.h"
+#include "parser/expression/abstract_expression.h"
+
+namespace terrier::parser {
+
+ConstantValueExpression::ConstantValueExpression(const type::TypeId type, std::unique_ptr<execution::sql::Val> value)
+    : ConstantValueExpression(type, std::move(value), nullptr) {
+  TERRIER_ASSERT(value_ != nullptr, "Didn't provide a value.");
+}
+
+ConstantValueExpression::ConstantValueExpression(const type::TypeId type, std::unique_ptr<execution::sql::Val> value,
+                                                 std::unique_ptr<byte> buffer)
+    : AbstractExpression(ExpressionType::VALUE_CONSTANT, type, {}),
+      value_(std::move(value)),
+      buffer_(std::move(buffer)) {
+  TERRIER_ASSERT(value_ != nullptr, "Didn't provide a value.");
+  TERRIER_ASSERT(value_->is_null_ || (type != type::TypeId::VARCHAR && type != type::TypeId::VARBINARY) ||
+                     (buffer_ == nullptr && GetValue().CastManagedPointerTo<execution::sql::StringVal>()->len_ <=
+                                                execution::sql::StringVal::InlineThreshold()) ||
+                     (buffer_ != nullptr && GetValue().CastManagedPointerTo<execution::sql::StringVal>()->len_ >
+                                                execution::sql::StringVal::InlineThreshold()),
+                 "Value should either be NULL, a non-varlen type, or varlen and below the threshold with no owned "
+                 "buffer, or varlen and above the threshold with a provided buffer.");
+}
+
+ConstantValueExpression &ConstantValueExpression::operator=(const ConstantValueExpression &other) {
+  if (this != &other) {  // self-assignment check expected
```

Comment:
&other or other? I'm not sure here.

Reply:
`&other` since `other` is a reference, while `this` is a pointer.